### PR TITLE
VP-6884: Verified status for an email address on the account blade

### PR DESCRIPTION
**Pull request by AliveMen**
Add permission for Verify Email switcher
Add Emeil verifivation event
Add switcher to ui and apply for permission

### CODE REVIEW

File: src/VirtoCommerce.Platform.Web/wwwroot/js/app/security/blades/account-detail.js [hunk 1]
```diff
@@ -44,6 +45,12 @@ angular.module('platformWebApp').controller('platformWebApp.accountDetailControl
             return !angular.equals(blade.currentEntity, blade.origEntity) && blade.hasUpdatePermission();
         }
 
+        blade.sendLink = function () {
+            if (blade.isLinkSent === false) {
```
Comment (eMazeika) [inline src/VirtoCommerce.Platform.Web/wwwroot/js/app/security/blades/account-detail.js:49]: 1. simplify to !blade.isLinkSent
2. set isLoading = true and lock UI from clicking second time

File: src/VirtoCommerce.Platform.Web/wwwroot/js/app/security/blades/account-detail.tpl.html [hunk 2]
```diff
@@ -56,15 +56,31 @@
...
+            <div class="form-label"> {{ 'platform.blades.account-detail.labels.verified' | translate }}
+            </div>
+            <div>
+                <label class="form-label __switch" >
+                    <input type="checkbox" ng-model="blade.currentEntity.emailConfirmed">
+                    <span class="switch" ng-class="{disabled: currentEntity.isReadOnly}"></span>
+                </label>
+            </div>
+            <div>
+                <a href="" ng-style="blade.isLinkSent && {'color':'green'} || {'color': ''}" ng-click="blade.sendLink()" va-permission="platform:security:verifyEmail">
```
Comment (eMazeika) [inline src/VirtoCommerce.Platform.Web/wwwroot/js/app/security/blades/account-detail.tpl.html:77]: remove one of the 2 identical va-permission usages

**Review by eMazeika — CHANGES_REQUESTED** [review]

File: src/VirtoCommerce.Platform.Web/wwwroot/js/app/security/resources/accounts.js [hunk 3]
```diff
@@ -12,6 +12,7 @@ angular.module('platformWebApp').factory('platformWebApp.accounts', ['$resource'
         lock: { url: 'api/platform/security/users/:id/lock', method: 'POST' },
         getUserApiKeys: { url: 'api/platform/security/users/:id/apikeys', method: 'GET', isArray: true},
         saveUserApiKey: { url: 'api/platform/security/users/apikeys', method: 'POST' },
-        deleteUserApiKey: { url: 'api/platform/security/users/apikeys', method: 'DELETE' }
+        deleteUserApiKey: { url: 'api/platform/security/users/apikeys', method: 'DELETE' },
+        verifyEmail: { url: 'api/platform/security/users/:id/verifyEmail', method: 'POST' }
```
Comment (Vectorfield4) [inline src/VirtoCommerce.Platform.Web/wwwroot/js/app/security/resources/accounts.js:16]: Why id here? API waiting for userName.

**Review by Vectorfield4 — CHANGES_REQUESTED** [review]

File: src/VirtoCommerce.Platform.Web/wwwroot/js/app/security/resources/accounts.js (hunk 3 above)
Comment (Vectorfield4) [reply src/VirtoCommerce.Platform.Web/wwwroot/js/app/security/resources/accounts.js:16]: image

File: src/VirtoCommerce.Platform.Web/wwwroot/js/app/security/blades/account-detail.tpl.html (hunk 2 above)
Comment (AliveMen) [reply src/VirtoCommerce.Platform.Web/wwwroot/js/app/security/blades/account-detail.tpl.html]: removed

File: src/VirtoCommerce.Platform.Web/Controllers/Api/SecurityController.cs [hunk 4]
```diff
@@ -662,6 +662,30 @@ public async Task<ActionResult<UserApiKey[]>> DeleteUserApiKeys([FromQuery] stri
...
         }
 
+        /// <summary>
+        /// Verify user email
+        /// </summary>
+        /// <param name="userName"></param>
+        [HttpPost]
+        [Route("users/{userName}/verifyEmail")]
+        [Authorize(PlatformConstants.Security.Permissions.SecurityVerifyEmail)]
+        public async Task<ActionResult> VerifyEmail([FromRoute] string userName)
```
Comment (ksavosteev) [inline src/VirtoCommerce.Platform.Web/Controllers/Api/SecurityController.cs:672]: better to change to SendVerificationEmail or TriggerVerification or someting like that
VerifyEmail is misleading

**Review by ksavosteev — CHANGES_REQUESTED** [review]

File: src/VirtoCommerce.Platform.Web/wwwroot/js/app/security/resources/accounts.js (hunk 3 above)
Comment (AliveMen) [reply src/VirtoCommerce.Platform.Web/wwwroot/js/app/security/resources/accounts.js]: renamed to userId in all places

File: src/VirtoCommerce.Platform.Web/Controllers/Api/SecurityController.cs (hunk 4 above)
Comment (AliveMen) [reply src/VirtoCommerce.Platform.Web/Controllers/Api/SecurityController.cs]: renamed

**Review by Vectorfield4 — APPROVED** [review]

**Review by eMazeika — APPROVED** [review]

**Review by ksavosteev — APPROVED** [review]

**Pull request merged**